net/dns/resolvd.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build openbsd
+// +build openbsd
+
+package dns
+
+import (
+	"os/exec"
+
+	"tailscale.com/types/logger"
+)
+
+func newResolvdManager(logf logger.Logf, interfaceName string) (*resolvdManager, error) {
+	return &resolvdManager{
+		logf:   logf,
+		ifName: interfaceName,
+	}, nil
+}
+
+// resolvdManager is an OSConfigurator which uses route(1) to teach OpenBSD's
+// resolvd(8) about DNS servers.
+type resolvdManager struct {
+	logf   logger.Logf
+	ifName string
+}
+
+func (m *resolvdManager) SetDNS(config OSConfig) error {
+	args := []string{
+		"nameserver",
+		m.ifName,
+	}
+
+	for _, s := range config.Nameservers {
+		args = append(args, s.String())
+	}
+
+	cmd := exec.Command("/sbin/route", args...)
```

Comment:
Is this constructed command safe to invoke repeatedly, with either the same nameservers or a different set? I assume so, but want to explicitly confirm since I don't have an OpenBSD box to test on.

Reply:
Yep, it adds/removes accordingly.

```
qbit@tal[0]:~$ cat /etc/resolv.conf                                                                                                                                                                         
nameserver 100.100.100.100 # resolvd: tun0
nameserver 10.6.0.2 # resolvd: em0
search qbit.github.beta.tailscale.net
qbit@tal[0]:~$ doas route nameserver tun0 100.100.100.100 
qbit@tal[0]:~$ cat /etc/resolv.conf                       
nameserver 100.100.100.100 # resolvd: tun0
nameserver 10.6.0.2 # resolvd: em0
search qbit.github.beta.tailscale.net
qbit@tal[0]:~$ doas route nameserver tun0 100.100.100.100 1.1.1.1 
qbit@tal[0]:~$ cat /etc/resolv.conf                               
nameserver 1.1.1.1 # resolvd: tun0
nameserver 100.100.100.100 # resolvd: tun0
nameserver 10.6.0.2 # resolvd: em0
search qbit.github.beta.tailscale.net
qbit@tal[0]:~$ doas route nameserver tun0 100.100.100.100         
qbit@tal[0]:~$ cat /etc/resolv.conf                       
nameserver 100.100.100.100 # resolvd: tun0
nameserver 10.6.0.2 # resolvd: em0
search qbit.github.beta.tailscale.net
qbit@tal[0]:~$ 
```